src/ui/mainwindow.cc

```diff
@@ -909,6 +928,13 @@ MainWindow::MainWindow( Config::Class & cfg_ ):
   if ( cfg.preferences.checkForNewReleases ) {
     QTimer::singleShot( 0, this, &MainWindow::checkNewRelease );
   }
+
+  timer = new QTimer( this );
+  connect( timer, &QTimer::timeout, this, [ this ]() {
+    saveStateData();
+  } );
+  //60 seconds
+  timer->start( 60000 );
```

Comment:
This is a wrong fix. I don't think anyone wants their dictionary software wants a background task that most of them unneeded 😅

Reply:
For pkill and systemd's shutting down specifically, we need to install a SIGTERM handler. https://github.com/xiaoyifang/goldendict-ng/pull/1728#issuecomment-2296502215

Reply:
Is there any way to handle the sigterm signal?

Reply:
https://doc.qt.io/qt-6/unix-signals.html